### PR TITLE
dev/financial#82 revert added setters

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
We added some structure to CRM_Core_Payment in the current unreleased code. However @artfulrobot has a proposal that @mattwire & I both agree is better architecture and less likely to break stuff.

I think the cleanest option here is to simply reverse these changes in 5.20 and do them again, better. I'd like the better to be in 5.20 but if we missed 5.20 I think it would not cause issues

Before
----------------------------------------
New but now deprecated methods in play

After
----------------------------------------
Slate cleaned to try again

Technical Details
----------------------------------------
@mattwire @artfulrobot

Comments
----------------------------------------

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@mattwire @artfulrobot - so merging this is priority 1 in my opinion - then we can finalised @artfulrobot's improved version - but we should get this merged before the rc is cut

File: api/v3/PaymentProcessor.php [hunk 1]
```diff
@@ -150,7 +139,7 @@ function civicrm_api3_payment_processor_pay($params) {
     }
     throw new API_Exception('Payment failed', $code, $errorData, $e);
   }
-  return civicrm_api3_create_success([$result], $params);
+  return civicrm_api3_create_success(array($result), $params);
```
Comment (mattwire) [inline api/v3/PaymentProcessor.php]: @eileenmcnaughton This shouldn't be in here

File: api/v3/PaymentProcessor.php [hunk 2]
```diff
@@ -160,7 +149,7 @@ function civicrm_api3_payment_processor_pay($params) {
  */
 function _civicrm_api3_payment_processor_pay_spec(&$params) {
   $params['payment_processor_id'] = [
-    'api.required' => TRUE,
+    'api.required' => 1,
```
Comment (mattwire) [inline api/v3/PaymentProcessor.php]: @eileenmcnaughton This shouldn't be in here

File: tests/phpunit/api/v3/PaymentProcessorTest.php [hunk 3]
```diff
@@ -154,8 +143,6 @@ public function testPaymentProcessorDelete() {
 
   /**
    * Check with valid params array.
-   *
-   * @throws \CRM_Core_Exception
```
Comment (mattwire) [inline tests/phpunit/api/v3/PaymentProcessorTest.php]: @eileenmcnaughton None of this needs to go?

File: api/v3/PaymentProcessor.php [hunk 4]
```diff
@@ -124,22 +124,11 @@ function _civicrm_api3_payment_processor_getlist_defaults(&$request) {
  *   API result array.
  *
  * @throws \API_Exception
- * @throws \CiviCRM_API3_Exception
  */
 function civicrm_api3_payment_processor_pay($params) {
-  /* @var CRM_Core_Payment $processor */
```
Comment (mattwire) [inline api/v3/PaymentProcessor.php]: @eileenmcnaughton Don't need to remove?

File: CRM/Core/Payment.php [hunk 5]
```diff
@@ -632,6 +257,8 @@ public function buildForm(&$form) {
    * @todo move to factory class \Civi\Payment\System (or similar)
    *
    * @param array $params
+   *
```
Comment (mattwire) [inline CRM/Core/Payment.php]: @eileenmcnaughton This doesn't need removing

File: CRM/Core/Payment.php [hunk 6]
```diff
@@ -81,7 +81,7 @@ abstract class CRM_Core_Payment {
     RECURRING_PAYMENT_END = 'END';
 
   /**
-   * @var array
+   * @var object
```
Comment (mattwire) [inline CRM/Core/Payment.php]: @eileenmcnaughton Don't think this should be in here

**Review by mattwire — CHANGES_REQUESTED** [review]
@eileenmcnaughton I think there's a few bits in here (mostly comments) that should not be.

**Comment by eileenmcnaughton** [comment]
@mattwire I've re-applied some of those changes in an extra commit - I think most of them are easy enough to re-spot / re-do

**Review by mattwire — APPROVED** [review]

**Comment by eileenmcnaughton** [comment]
Thanks @mattwire  - @artfulrobot - where are we at with your alternative - it's still marked WIP but I think you possibly think it is review-ready

**Pull request merged**